java/src/jmri/jmrix/jinput/TreeModel.java

```diff
@@ -73,10 +76,14 @@ DefaultMutableTreeNode insertNode(DefaultMutableTreeNode pChild, DefaultMutableT
         }
         // represent this one
         index = pParent.getChildCount();
-        insertNodeInto(pChild, pParent, index);
+        try {
+            insertNodeInto(pChild, pParent, index);
+        } catch (IllegalArgumentException e) {
```

Comment:
I'm not sure that this is actually the right way to suppress they error you were seeing during a test.

The question is WHY is the node being inserted an illegal argument?  i.e. how did we get to the point where we call the insert with a node that isn't right, and can that situation occur in practice.

Reply:
The only "IllegalArgumentException" is trying to insert a child into a parent when the child is an ancestor of the parent… How that happens is anyone's guess… the important thing is that the child already exist in the tree (which the code following the try/catch tests for). Weird, but I didn't write it.

Reply:
Does it happen in practice, or only during testing?

Reply:
I actually don't know about testing… other than all test pass (other than the weird help one).
I suspect that this code just wasn't getting used… all the external system connections that use USB actually use Serial over USB (via the system's serial (or comm) ports). This PR implements an external system connection that actually uses USB. So definitely in practice… now… ;-)

Reply:
Ok, the problem is that getNode keyed off of the component… but for controllers that's always null… so one node for all controllers... not good. If you key off of the controller instead the that's the same for all components... again, not good. The fix is to key off the name; which seems to be unique between all controllers and components… (except for "UNKNOWN [UNKNOWN]"'s)… but I'm thinking that those should be ignored anyway...)
Just committed a fix… everything's tested and works as expected.